BDArmory/ModuleWeapon.cs

```diff
@@ -1604,42 +1592,21 @@ void Aim()
 
 
             //aim assist
-            Vector3 finalTarget = targetPosition;
-            Vector3 originalTarget = targetPosition;
-            targetDistance = Vector3.Distance(finalTarget, transform.position);
-            targetLeadDistance = targetDistance;
+            targetLeadDistance = Vector3.Distance(targetPosition, transform.position);
+            fixedLeadOffset = Vector3.zero;
+            finalAimTarget = targetPosition;
+            FiringSolutionVector = null;
+
             if ((BDArmorySettings.AIM_ASSIST || aiControlled) && eWeaponType != WeaponTypes.Laser)
             {
-                float gAccel = (float)FlightGlobals.getGeeForceAtPosition(finalTarget).magnitude;
-                float time = targetDistance / (bulletVelocity);
-
                 if (targetAcquired)
                 {
-                    float time2 = VectorUtils.CalculateLeadTime(finalTarget - fireTransforms[0].position,
-                        targetVelocity - vessel.Velocity(), bulletVelocity);
-                    if (time2 > 0) time = time2;
-                    finalTarget += (targetVelocity - vessel.Velocity()) * time;
-                    
-                    //target vessel relative velocity compensation
-                    Vector3 acceleration = targetAcceleration;
-                    finalTarget += (0.5f * acceleration * time * time); //target acceleration
+                    FiringSolutionVector = calculateFiringSolution(targetPosition, targetVelocity - vessel.Velocity(), targetAcceleration);
                 }
                 else if (vessel.altitude < 6000)
                 {
-                    float time2 = VectorUtils.CalculateLeadTime(finalTarget - fireTransforms[0].position,
-                        -part.rb.velocity, bulletVelocity);
-                    if (time2 > 0) time = time2;
-                    finalTarget += (-part.rb.velocity * (time + Time.fixedDeltaTime));
-                    //this vessel velocity compensation against stationary
+                    FiringSolutionVector = calculateFiringSolution(targetPosition, -(part.rb.velocity + Krakensbane.GetFrameVelocityV3f()), Vector3.zero);
                 }
-                Vector3 up = (finalTarget - vessel.mainBody.transform.position).normalized;
-                if (bulletDrop && vessel.srfSpeed < 750)
-                    finalTarget += (0.5f * gAccel * time * time * up); //gravity compensation
-
-                targetLeadDistance = Vector3.Distance(finalTarget, fireTransforms[0].position);
-
-                fixedLeadOffset = originalTarget - finalTarget; //for aiming fixed guns to moving target	
-
```

Comment:
Moved trajectory calculations to a separate method - calculateFiringSolution. Well, actually deleted these and wrote new ones, not moved.